examples/simple_reverse.coffee

```diff
@@ -0,0 +1,76 @@
+# This is a simple example of how to use the slack-client module in CoffeeScript. It creates a
+# bot that responds to all messages in all channels it is in with a reversed
+# string of the text received.
+#
+# To run, copy your token below, then, from the project root directory:
+#
+# To run the script directly
+#    npm install
+#    node_modules/coffee-script/bin/coffee examples/simple_reverse.coffee 
+#
+# If you want to look at / run / modify the compiled javascript
+#    npm install
+#    node_modules/coffee-script/bin/coffee -c examples/simple_reverse.coffee 
+#    cd examples
+#    node simple_reverse.js
+#
+
+Slack = require '..'
+
+token = 'xoxb-YOUR-TOKEN-HERE' # Add a bot at https://my.slack.com/services/new/bot and copy the token here.
+autoReconnect = true
+autoMark = true
+
+slack = new Slack(token, autoReconnect, autoMark)
+
+slack.on 'open', ->
+  channels = []
+  groups = []
+  unreads = slack.getUnreadCount()
+
+  # Get all the channels that bot is a member of
+  channels = ("##{value.name}" for key, value of slack.channels when slack.channels[key].is_member)
+  # Get all groups that are open and not archived 
+  groups = (for key, value of slack.groups
+    if slack.groups[key].is_open and not slack.groups[key].is_archived
+      value.name)
```

Comment:
Just noticed that this expanded loop, as opposed to the comprehension, seems to be adding an "void 0" to the end of the groups array.

I can prevent this with an empty *else* block, but that seems gross.  Is there a nicer way?

```
groups = (function() {
  var ref, results;
  ref = slack.groups;
  results = [];
  for (id in ref) {
    group = ref[id];
    if (group.is_open && !group.is_archived) {
      results.push(group.name);
    } else {
      results.push(void 0);
    }
  }
  return results;
})();
```